lib/Runtime/Language/InlineCache.cpp

```diff
@@ -697,7 +674,7 @@ namespace Js
 #endif
             if (!PHASE_OFF1(Js::CloneCacheInCollisionPhase))
             {
-                if (!inlineCaches[inlineCacheIndex].IsEmpty() && !inlineCaches[inlineCacheIndex].NeedsToBeRegisteredForStoreFieldInvalidation())
+                if (!inlineCaches[inlineCacheIndex].IsEmpty() && !inlineCaches[inlineCacheIndex].NeedsToBeRegisteredForStoreFieldInvalidation() && GetSize() != 1)
```

Comment:
It's really about the specific size value 1 rather than the abstract variable. Point is if there is only 1 slot, we can't clone to empty slot.